Mathlib/NumberTheory/FactorisationProperties.lean

```diff
@@ -170,21 +170,24 @@ theorem infinite_deficient : {n : ℕ | n.Deficient}.Infinite := by
   rw [Set.infinite_iff_exists_gt]
   intro a
   obtain ⟨b, h1, h2⟩ := exists_infinite_primes a.succ
-  exact ⟨b, Set.mem_setOf.mpr (prime_deficient h2), h1⟩
+  exact ⟨b, Set.mem_setOf.mpr (Prime.deficient h2), h1⟩
 
 theorem infinite_even_deficient : {n : ℕ | Even n ∧ n.Deficient}.Infinite := by
+  rw [Set.infinite_iff_exists_gt]
+  intro n
   use 2 ^ (n + 1)
   constructor
+  · exact Set.mem_sep (Set.mem_def.mpr ⟨2 ^ n, by ring⟩) (Prime.deficient_pow prime_two)
```

Comment:
No need to use all those definitional `Set` lemmas when providing a proof term
```suggestion
  · exact ⟨2 ^ n, by ring, prime_two.deficient_pow⟩
```

Reply:
This gives an error.

Reply:
Do you want me to remove the set lemmas? I think I need at least `Set.mem_sep` to prove it

Reply:
Ah probably
```suggestion
  · exact ⟨⟨2 ^ n, by ring⟩, prime_two.deficient_pow⟩
```
instead

Reply:
Thank you! I didn't know you could do that for sets.